packages/dask-task-models-library/src/dask_task_models_library/container_tasks/protocol.py

```diff
@@ -17,20 +20,89 @@
 ContainerLabelsDict: TypeAlias = dict[DockerLabelKey, str]
 
 
+class TaskOwner(BaseModel):
+    user_id: UserID
+    project_id: ProjectID
+    node_id: NodeID
+
+    parent_project_id: ProjectID | None
+    parent_node_id: NodeID | None
```

Comment:
If the job is started through the api-server, how do you get this info? There, one passes the `job_id` of the job one wants to start, which I guess corresponds to the `project_id` here. But how would the parent info have to be passed?

Reply:
so currently this mostly follows the current workflow that @mguidon is using.
The parent _project_id_ or _job_id_ is defined in the _projects_metadata_ DB table via the _PATCH /jobs/{solver_key}/releases/{version}/jobs/{id}/metadata_ entrypoint on the api-server.
**If and only if** there is a _node_id_ defined in these metadata does the webserver then transfers the parent ids to the director-v2, and so on...

Reply:
Aha, ok thanks for the clarification. I had forgotten about that PATCH endpoint in the API server